app/javascript/mastodon/components/status_list.js

```diff
@@ -53,6 +60,39 @@ class StatusList extends ImmutablePureComponent {
 
   componentWillUnmount () {
     this.detachScrollListener();
+    this.detachIntersectionObserver();
+  }
+
+  attachIntersectionObserver () {
+    const onIntersection = (entries) => {
+      const isIntersecting = { };
+
+      entries.forEach(entry => {
+        const statusId = entry.target.getAttribute('data-id');
+
+        isIntersecting[statusId] = entry.isIntersecting;
+      });
+
+      this.setState(state => ({
+        isIntersecting: Object.assign({ }, state.isIntersecting, isIntersecting),
+      }));
+    };
+
+    const options = {
+      root: this.node,
+      rootMargin: '100% 0px',
+    };
```

Comment:
> Whereas if we allowed for 5x the available viewport height, then that should lead to less jumpy scrolling because it won't start un-rendering until you've scrolled 5x of the way down.

Don't think this is how it works. With 5x it does the following:

```
Not rendered - next to render when you scroll up
Rendered
Rendered
Rendered
Rendered
Rendered
---------- <- Viewport start
Rendered
---------- <- Viewport end
Rendered
Rendered
Rendered
Rendered
Rendered
Not rendered - next to render when you scroll down
```

The same amount of rendering happens when you scroll, it is just that the chance of viewing the fade in animation when you scroll fast is smaller.